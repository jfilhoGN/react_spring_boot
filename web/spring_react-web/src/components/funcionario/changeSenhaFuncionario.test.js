import React from 'react';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import ChangeSenhaFuncionario from './changeSenhaFuncionario';

describe('changeSenhaFuncionario', () => {
    test('render component', () => {
        const wrapper = shallow(<ChangeSenhaFuncionario />);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });
    
    test('values of state', () => {
        const wrapper = shallow(<ChangeSenhaFuncionario/>)
        expect(wrapper.state).toMatchSnapshot();
    });
    
    test('getFuncionario', () => {
        //arrumar
    });

    test('input text senhaAntiga', () => {
        const wrapper = shallow(<ChangeSenhaFuncionario />);
        wrapper.find('#ioldSenha').simulate('change', { target: { value: '123456'} });
        expect(wrapper.state().senhaAtual).toEqual('123456');
    });
    
    test('input text novaSenha', () => {
        const wrapper = shallow(<ChangeSenhaFuncionario />);
        wrapper.find('#inewSenha').simulate('change',{ target: { value: '12345'} });
        expect(wrapper.state().novaSenha).toEqual('12345');
    });

    test('input text confirmSenha', () => {
        const wrapper = shallow(<ChangeSenhaFuncionario />);
        wrapper.find('#iconfirmSenha').simulate('change',{ target: { value: '12345'} });
        expect(wrapper.state().confirmSenha).toEqual('12345');
    });

    /* const flushPromises = () => new Promise(resolve => setImmediate(resolve));
    test('button changeSenha', async () => {
        const onButtonClick = sinon.spy();
        global.sessionStorage.getItem.mockReturnValue('{"id": "28A5B171C9FF4FBFB4BC0481C0BD0435", "token":"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiIyOEE1QjE3MUM5RkY0RkJGQjRCQzA0ODFDMEJEMDQzNSJ9.T4K4br5kaEboh4UIl_gkhYi3FcMD1Wk_bIcQWVv-tRc"}')
        const wrapper = shallow(<ChangeSenhaFuncionario />);
        await flushPromises();
        wrapper.find('.updateSenha').simulate('click', { preventDefault() {} });
        expect(onButtonClick.notCalled).toBe(true);
    }) */

    test('submitSenha()', () => {
        //arrumar
    });

    test('sendSenha', () => {
        //arrumar
    });

    
    
});
