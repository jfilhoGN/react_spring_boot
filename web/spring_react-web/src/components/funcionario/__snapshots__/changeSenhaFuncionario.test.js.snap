// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeSenhaFuncionario render component 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div>
      <Row>
        <Col>
          <form
            className="col s13"
          >
            <Input
              checked={false}
              id="ioldSenha"
              label="Senha Antiga"
              name="ioldSenha"
              onChange={[Function]}
              s={12}
              type="password"
            />
            <Input
              checked={false}
              id="inewSenha"
              label="Nova Senha"
              name="inewSenha"
              onChange={[Function]}
              s={12}
              type="password"
            />
            <Input
              checked={false}
              id="iconfirmSenha"
              label="Confirmar nova senha"
              name="iconfirmSenha"
              onChange={[Function]}
              s={12}
              type="password"
            />
            <Button
              className="updateSenha btn button-espaco waves-effect waves-light btn-small grey darken-3"
              node="button"
              onClick={[Function]}
              type="submit"
            >
              <i
                className="material-icons"
              >
                update
              </i>
            </Button>
            <Button
              className="btn button-espaco waves-effect waves-light btn-small grey darken-3"
              node="button"
              onClick={[Function]}
              type="submit"
            >
              <i
                className="material-icons"
              >
                close
              </i>
            </Button>
          </form>
        </Col>
      </Row>
    </div>
  </div>
  <div
    className="card-reveal"
  >
    <span
      className="card-title grey-text text-darken-4"
    >
      <Icon
        right={true}
      >
        close
      </Icon>
    </span>
  </div>
</div>
`;

exports[`changeSenhaFuncionario values of state 1`] = `[Function]`;
